src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ILockStatement.cs

```diff
@@ -1025,5 +1303,747 @@ void M(P input, bool b)
 
             VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(compilation, expectedGraph, expectedDiagnostics);
         }
+
+        [Fact, CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        public void LockFlow_LockObject()
+        {
+            var source = """
+                using System;
+                using System.Threading;
+
+                class C
+                {
+                    void M()
+                    /*<bind>*/{
+                        Lock l = new Lock();
+                        lock (l)
+                        {
+                        }
```

Comment:
It will also be useful to test scenarios with non-empty body. For error scenarios as well.